pandas/core/arrays/masked.py

```diff
@@ -1097,6 +1103,30 @@ def _wrap_reduction_result(self, name: str, result, skipna, **kwargs):
             return self._maybe_mask_result(result, mask)
         return result
 
+    def _wrap_min_count_reduction_result(
+        self, name: str, result, skipna, min_count, **kwargs
+    ):
+        if min_count == 0 and isinstance(result, np.ndarray):
+            return self._maybe_mask_result(result, np.zeros(result.shape, dtype=bool))
+        return self._wrap_reduction_result(name, result, skipna, **kwargs)
+
+    def _wrap_na_result(self, name):
+        mask = np.ones(1, dtype=bool)
+
+        if self.dtype.kind == "f":
+            np_dtype = "float64"
+        elif name in ["mean", "median", "var", "std", "skew"]:
+            np_dtype = "float64"
+        elif self.dtype.kind == "i":
+            np_dtype = "int64"
+        elif self.dtype.kind == "u":
+            np_dtype = "uint64"
+        else:
+            raise TypeError(self.dtype)
```

Comment:
Should  corr, sem, and kurt be part of the first elif? And is this not hit by e.g. a sum of Boolean?

In general I'm not a fan of encoding op-specific logic like this. It can lead to bugs if behavior is changed in one place but not the other.

Reply:
Yeah, good points, I'll have to look into this again to see if I can do it differently.

Reply:
Wow - nice! So this wasn't necessary?

Reply:
yeah It was needed in the first draft, then I changed the code making it unnecessary.

There is still some things missing in this PR. I'll update today.